meta-mender-toradex-nxp/conf/machine/colibri-imx7-mender.conf

```diff
@@ -25,7 +25,7 @@ BOOTENV_SIZE ?= "0x20000"
 BOOTENV_RANGE ?= "0x40000"
 
 # 512MB of NAND flash
-MENDER_STORAGE_TOTAL_SIZE_MB ?= "512"
+MENDER_STORAGE_TOTAL_SIZE_MB = "512"
```

Comment:
I'm ok with the patch, but I have a suggestion: Leave this alone, and do this patch instead:

```
diff --git a/meta-mender-demo/conf/layer.conf b/meta-mender-demo/conf/layer.conf
index 409c5e8..1deb9c8 100644
--- a/meta-mender-demo/conf/layer.conf
+++ b/meta-mender-demo/conf/layer.conf
@@ -12,7 +12,7 @@ BBFILE_COLLECTIONS += "mender-demo"
 BBFILE_PATTERN_mender-demo = "^${LAYERDIR}/"
 BBFILE_PRIORITY_mender-demo = "6"
 
-MENDER_STORAGE_TOTAL_SIZE_MB ?= "608"
+MENDER_STORAGE_TOTAL_SIZE_MB_DEFAULT ?= "608"
 IMAGE_OVERHEAD_FACTOR = "1.0"
 
 LAYERSERIES_COMPAT_mender-demo = "sumo"
```

That would give you the best of both worlds: It would allow the demo layer to be overridden, and allow the size in `colibri-imx7-mender.conf` to be overridden by the user as well.